opentelemetry/src/context.rs

```diff
@@ -305,21 +308,16 @@ impl Context {
         }
     }
 
-    pub(super) fn current_with_trace_context(
-        value: super::trace::context::SynchronizedSpan,
-    ) -> Self {
+    pub(super) fn current_with_span_internal(value: SynchronizedSpan) -> Self {
```

Comment:
Why the `_internal` suffix?

Reply:
I thought it better to distinguish them from the `current_with_span` and `with_span` methods on the `TraceContextExt` trait, which this struct also implements.  "internal" denotes they are only relevant internal to this crate, since the `SyncrhonizedSpan` method they consume is not a public type.

Reply:
Would the name `with_synchronized_span` better?  The disambiguation syntax feels a bit more unnatural (to me), as it affects current callsites of the existing functions, not the new ones, but I could be convinced to get used to it, if that is the consensus.

As an example, here is a before/after:
```rust
        let cx = Context::new().with_span(NoopSpan {
            span_context: valid_span_context(),
        });          
```
becomes
```rust
        let cx = TraceContextExt::with_span(
            &Context::new(),
            NoopSpan {
                span_context: valid_span_context(),
            },
        );
```

Reply:
Well, you can write it more like `<Context as TraceContextExt>::with_span()` which is slightly less verbose (though maybe also less familiar? I do think `with_synchronized_span()` is a pretty good alternative.

Reply:
Renamed it.  Thanks.